### DISCUSSION TITLE
Is there a way to test for incrementing/non-duplicate entity IDs created during testing?

### DISCUSSION THREAD

**Discussion by hydroculator** [schemathesis/schemathesis#2351] [Q&A]
We ran into an issue where we were returning 0 for all newly created objects of a certain type. Is there a way to get Schemathesis to detect this? That all the objects created have unique returned IDs?

Thanks

**Comment by Stranger6667** [comment]
Hi!

I think you can use a [custom check](https://schemathesis.readthedocs.io/en/stable/extending.html#checks) for it:

```python
import schemathesis

SEEN_IDS = set()

@schemathesis.check
def only_unique_ids(response, case) -> None:
    id = response.json().get("id")
    if id in SEEN_IDS:
        raise AssertionError(f"Duplicate ID: {id}")
    SEEN_IDS.add(id)
```

Something like this :) Here is a [section](https://schemathesis.readthedocs.io/en/stable/extending.html#enabling-extensions) on how to enable extensions
